### PR TITLE
chore: Adds sleep in tests to avoid out of capacity errors

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds sleep in tests to avoid out of capacity errors. 

This PR is to test the hypothesis that out of capacity errors in tests are caused because clusters are created simultaneously in the same project, see [HELP-65223](https://jira.mongodb.org/browse/HELP-65223) for more info.

The previous PR didn't work because the way tests are executed in Go. They run in serial until `t.Parallel()` is called. In that PR sleep was before that call so only test preparation was serialized but then all parallel tests would start at the same time.

In this PR, the sleep is in PreCheck functions, that are called when the tests are already running in parallel.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/testutil/acc/shared_resource.go [hunk 1]
```diff
@@ -48,8 +48,6 @@ func ProjectIDExecution(tb testing.TB) string {
 		sharedInfo.projectName = RandomProjectName()
 		tb.Logf("Creating execution project: %s\n", sharedInfo.projectName)
 		sharedInfo.projectID = createProject(tb, sharedInfo.projectName)
-	} else {
```
Comment (lantoli) [inline internal/testutil/acc/shared_resource.go:51]: revert previous PR

File: GNUmakefile [hunk 2]
```diff
@@ -8,7 +8,7 @@ endif
 
 ACCTEST_REGEX_RUN?=^TestAcc
 ACCTEST_TIMEOUT?=300m
-PARALLEL_GO_TEST?=20
+PARALLEL_GO_TEST?=50
```
Comment (lantoli) [inline GNUmakefile:11]: don't limit parallel tests

File: internal/testutil/acc/shared_resource.go [hunk 3]
```diff
@@ -87,10 +85,23 @@ func ClusterNameExecution(tb testing.TB) (projectID, clusterName string) {
...
+// SerialSleep waits a few seconds so clusters are not created concurrently in the execution project, see HELP-65223.
+// This must be called once the test is marked as parallel, e.g. in PreCheck inside Terraform tests.
+func SerialSleep(tb testing.TB) {
+	tb.Helper()
+	SkipInUnitTest(tb)
+	require.True(tb, sharedInfo.init, "SetupSharedResources must called from TestMain test package")
+
+	sharedInfo.muSleep.Lock()
+	defer sharedInfo.muSleep.Unlock()
+	time.Sleep(10 * time.Second)
```
Comment (EspenAlbert) [inline internal/testutil/acc/shared_resource.go:97]: Would it be good to have some jitter?

Comment (EspenAlbert) [inline internal/testutil/acc/shared_resource.go:97]: This means parallel test N will be delayed by 10s x N? Why 10s?

Comment (lantoli) [reply internal/testutil/acc/shared_resource.go:97]: correct. 10s to pick a number that is big enough to make sure clusters are not running very close. we're not still sure if this is the issue.

once we validate this is the problem (or not), then we can get the number lower

Comment (EspenAlbert) [reply internal/testutil/acc/shared_resource.go:97]: Okay. Thanks for explaining and investigating this 👍

**Review by EspenAlbert — APPROVED** [review]

**Review by marcosuma — APPROVED** [review]

**Pull request merged**